pkg/kontainer-engine/drivers/aks/aks_driver.go

```diff
@@ -796,26 +805,20 @@ func (d *Driver) createOrUpdate(ctx context.Context, options *types.DriverOption
 			osDiskSizeGBPointer = to.Int32Ptr(int32(driverState.AgentOsdiskSizeGB))
 		}
 
-		agentStorageProfile := containerservice.ManagedDisks
-		if driverState.AgentStorageProfile != "" {
-			agentStorageProfile = containerservice.StorageProfileTypes(driverState.AgentStorageProfile)
-		}
-
-		agentVMSize := containerservice.StandardD1V2
```

Comment:
Why are we no longer setting this? Was it removed from the API?

Reply:
Correct. The type still exists, but it is no longer a field in the `ManagedClusterAgentPoolProfile`.

Reference: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2020-09-01/containerservice#ManagedClusterAgentPoolProfile

Reply:
Thinking about this a little more, I'm wondering if we should deprecate `agentStorageProfile` since the API has done so with the `ManagedClusterAgentPoolProfile`. Thoughts @rmweir?

Reply:
@nickgerace if we don't use it for anything else than yes. I think I may have worked on this in the past; users wanted to be able to use managed disk storage. I was told in a kubernetes PR for the same issue that the managed disk was going to become the only option I believe.

Reply:
You'll want to create a UI issue for this so the option can be removed.

Reply:
Created issue: https://github.com/rancher/rancher/issues/29908